File: utils/tree.cpp
#include "tree.h"

bool is_same_tree(TreeNode* t1, TreeNode* t2)
{
    if (!t1 && !t2)
        return true;
    if (!t1 || !t2)
        return false;
    if (t1->val != t2->val)
        return false;
    return is_same_tree(t1->left, t2->left) && is_same_tree(t1->right, t2->right);
}

File: utils_py/tree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def is_same_tree(t1, t2):
    if not t1 and not t2:
        return True
    if not t1 or not t2:
        return False
    if t1.val != t2.val:
        return False
    return is_same_tree(t1.left, t2.left) and is_same_tree(t1.right, t2.right)
